Add `graphics` module

dwursteisen opened the pull request:
Add `graphics` module which add ShapeRenderer extension methods.

czyzby commented on CHANGELOG.md:
```diff
@@ -1,5 +1,7 @@
 #### 1.9.8-SNAPSHOT
 
+- **[FEATURE]** (`tx-graphics`) Add new graphics module which add graphics utilities extension methods.
```
`tx-graphics` -> `ktx-graphics`

czyzby replied:
The very first changelogs do not contain that, but currently when we're adding a new module, we tend to list all added features in a sublist.

```markdown
* **[FEATURE]** (`ktx-graphics`) Add new graphics module which add graphics utilities extension methods.
  * Added `xyz` extension method.
  (...)
```

czyzby commented on graphics/README.md:
```diff
@@ -0,0 +1,19 @@
+
+### Guide
+
+#### ShapeRenderer
```
`` `ShapeRenderer` ``

czyzby commented on graphics/README.md:
```diff
@@ -0,0 +1,19 @@
+#### ShapeRenderer
+
+Add methods to use `ShapeRenderer` class with `Vector2` and `Vector3` classes
+when LibGDX exposes methods with x, y or z args.
```
Can you list all extension methods in the readme?

czyzby commented on graphics/gradle.properties:
```diff
@@ -0,0 +1,2 @@
+projectName=ktx-graphics
+projectDesc=Kotlin utilities for handling graphics in LibGDX applications.
```
Missing new line at the end of the file.

czyzby commented on graphics/src/main/kotlin/ktx/graphics/graphics.kt:
```diff
@@ -0,0 +1,93 @@
+/** Draws a cube using {@link ShapeType#Line} or {@link ShapeType#Filled}. The pos specify the bottom, left, front corner
+ * of the rectangle. */
+fun ShapeRenderer.box(pos: Vector3, width: Float, height: Float, depth: Float) {
+  this.box(pos.x, pos.y, pos.z, width, height, depth)
```
`this.` is redundant.

Does it make sense to add any defaults for the parameters?

czyzby commented on graphics/src/main/kotlin/ktx/graphics/graphics.kt:
```diff
@@ -0,0 +1,93 @@
+/** Draws a circle using {@link ShapeType#Line} or {@link ShapeType#Filled}. */
+fun ShapeRenderer.circle(pos: Vector2, radius: Float, segment: Int) {
+  this.circle(pos.x, pos.y, radius, segment)
+}
```
Can this and the previous functions be merged with a default argument for `segment`?

czyzby commented on graphics/src/main/kotlin/ktx/graphics/graphics.kt:
```diff
@@ -0,0 +1,93 @@
+/** Multiplies the current transformation matrix by a scale matrix. */
+fun ShapeRenderer.scale(scale: Vector3) {
+  this.scale(scale.x, scale.y, scale.z)
+}
```
There are no tests for this module. Is there any way you could test these functions? Since `ShapeRenderer` "output" is not trivial to test, you could just use a mocked renderer and check if the original methods are called with the expected parameters.

czyzby commented:
I'm assuming you don't have time to write the tests, so I'm going to merge this module as-is and apply the requested changes myself before publishing the snapshot release. I might also move some utilities from `ktx-app` to `ktx-graphics`, we'll see. Thanks for the contribution.

dwursteisen commented:
Hello

Sorry for this late reply but

1) I totally forgot about this pull request. Lot of things was happening
lately.

2) I don't know if I pushed my last changes (I think I did it one month
ago, it was mostly about removing "this" etc)

3) I'm in vacation right know. And stuck in traffic. So I haven't access to
a computer right now 😅

I'll try to check this in few hours. Without response from my side, conside
it as ok.

And again sorry, it's slip out of my mind

Regards

Le dim. 29 avr. 2018 à 22:19, MJ <notifications@github.com> a écrit :

> @dwursteisen <https://github.com/dwursteisen> I moved some utilities from
> ktx-app, added tests for the ShapeRenderer extensions and applied the
> change requests from the review. Let me know what you think. I plan on
> releasing next version this week with ktx-graphics; snapshot is being
> uploaded as I'm writing this.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/libktx/ktx/pull/138#issuecomment-385251031>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAWxaT3VXtg3FBewXmAaDCnLSrw2RWcHks5ttb3mgaJpZM4S9zmZ>
> .
>

czyzby commented:
@dwursteisen It's OK, don't worry about it, enjoy your vacation. :)

czyzby commented:
I'm going to assume the `ktx-graphics` module is fine for now. Feel free to open an issue or comment here if you think anything needs improvement.
